zerver/views/registration.py

```diff
@@ -303,6 +310,9 @@ def accounts_register(request: HttpRequest) -> HttpResponse:
             bulk_add_subscriptions([realm.signup_notifications_stream], [user_profile])
             send_initial_realm_messages(realm)
 
+            # On realm creation PreregistrationUser object is always set as 1
+            do_set_confirmation_status(key, confirmation.type)
```

Comment:
This is not required. We are currently setting status to `STATUS_ACTIVE` to record that the user clicked the link. That is already taken care of by `check_prereg_key_and_redirect`